### PR TITLE
Covid19 survey

### PR DESCRIPTION
**Pull request by gangooteli**
# Description
This contains jupyter notebook to get the feature contirbution of 'Total Number of COVID Infected' in a Panchayat on covid19 survey dataset using SHAP library

User story # (US link)
https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/1334

# How Has This Been Tested?
The PR contains Jupyter notebook and test cases has not been added.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests
- [ ] New and existing unit tests pass locally with my changes
- [x] My Code is review by 1 People

### CODE REVIEW

File: src/analyses/EDA/covid19_survey_analysis/constants.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    "What facilities are available for returned Migrants in your Panchayat?--OTHER--",
+    "What activities have been initiated in your Panchayat for safety from COVID-19?--OTHER--",
+    "What are the reasons?--OTHER--",
+    "Are these community committees there in your Panchayat?--OTHER--",
+    "What activities are going on in your panchayat to provide livelihoods to returned migrant community?--OTHER--",
+    "Under MNREGA, what activities are going on in your panchayat to provide a livelihood to the returned migrant community?--OTHER--",
+    "What kind of Collaboration is desired?--OTHER--",
+]
+
+option_ques = ["--OPTION--|Pradhanmantri Kisan Samman Yojana",
```
Comment (markus-hinsche) [inline src/analyses/EDA/covid19_survey_analysis/constants.py:32]: What is `option_ques`?
What is `Pradhanmantri Kisan Samman Yojana`?

File: src/analyses/EDA/covid19_survey_analysis/data_exploration.ipynb [hunk 2]
```diff
@@ -0,0 +1,1698 @@
+{
```
Comment (markus-hinsche) [inline src/analyses/EDA/covid19_survey_analysis/data_exploration.ipynb:1]: <pre class="ql-syntax" spellcheck="false">fans_survey_data = 'covid19-data/Panchayat Survey-FaNS-MGSA(10.05.21).xlsx'
dmks_survey_Dat = 'covid19-data/Panchayat Survey_ DMKS (10.05.21).xlsx'
</pre><p> </p><p>Where is this data? If I would like to check that this notebook works, I would need this data. </p><p>Could we add instructions on how to access this data so we can run this notebook?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Welthungerhilfe/cgm-ml/pull/354/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (markus-hinsche) [inline src/analyses/EDA/covid19_survey_analysis/data_exploration.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    dmks_sheet_name = fans_sheet_name = 'Raw Data'
</pre><p>Why do we put this into 2 variables?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Welthungerhilfe/cgm-ml/pull/354/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

**Comment by markus-hinsche** [comment]
.github/pull_request_template.md provides a PR template. Could we fill this to express the status of this PR's code?

File: src/analyses/EDA/covid19_survey_analysis/constants.py (hunk 1 above)
Comment (gangooteli) [reply src/analyses/EDA/covid19_survey_analysis/constants.py:33]: Pradhanmantri Kisan Samman Yojana is a govt program to ensure minimum income support of farmers and help farmers in India.

File: src/analyses/EDA/covid19_survey_analysis/data_exploration.ipynb (hunk 2 above)
Comment (gangooteli) [reply src/analyses/EDA/covid19_survey_analysis/data_exploration.ipynb:1]: Added PoC for dataset in Readme

Comment (gangooteli) [reply src/analyses/EDA/covid19_survey_analysis/data_exploration.ipynb:1]: Fixed it.

**Comment by gangooteli** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/Welthungerhilfe/cgm-ml/pull/354/discussion/) on ReviewNB
> 
> _markus-hinsche commented on 2021-06-03T17:41:12Z_
> ----------------------------------------------------------------
> 
> Line #1.    import shap
> 
> Do we have to install shap? Could we add setup instructions so this will not throw an unexpected error

Added installation of shap in notebook.

**Comment by gangooteli** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/Welthungerhilfe/cgm-ml/pull/354/discussion/) on ReviewNB
> 
> _markus-hinsche commented on 2021-06-03T17:41:16Z_
> ----------------------------------------------------------------
> 
> What is the result?
> 
> We prepared data, we build a model, but what have we learned?
> 
> Can we summarize our thoughts/interpretation of results in a markdown cell with text?

Added in the notebook

> View / edit / reply to [this conversation](https://app.reviewnb.com/Welthungerhilfe/cgm-ml/pull/354/discussion/) on ReviewNB
> 
> _markus-hinsche commented on 2021-06-03T17:41:14Z_
> ----------------------------------------------------------------
> 
> Why do we use a random forest? Could we add some markdown text to motivate this?

Added in the notebook

File: src/analyses/EDA/covid19_survey_analysis/data_exploration.ipynb (hunk 2 above)
Comment (markus-hinsche) [reply src/analyses/EDA/covid19_survey_analysis/data_exploration.ipynb:1]: <p>I can't see this fix</p>

**Review by markus-hinsche — APPROVED** [review]
LGTM

Small caveat: I can't see the fix for the double assignment `dmks_sheet_name = fans_sheet_name = 'Raw Data'`

**Comment by gangooteli** [comment]
> LGTM
> 
> Small caveat: I can't see the fix for the double assignment `dmks_sheet_name = fans_sheet_name = 'Raw Data'`

image

Fixed it already. Attaching the above image

**Pull request merged**